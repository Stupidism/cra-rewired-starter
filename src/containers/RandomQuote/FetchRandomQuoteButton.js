import React from 'react';
import { Button } from 'antd';

import connect from 'store/connect';
import {
  fetchRandomQuote,
  isRandomQuoteLoading,
} from 'store/modules/randomQuote';

import './RandomQuote.less';

export const FetchRandomQuoteButton = ({ loading, onLoad, children }) => (
  <Button type="primary" onClick={onLoad} loading={loading}>
    {children}
  </Button>
);

export default connect(
  {
    loading: isRandomQuoteLoading,
  },
  {
    onLoad: fetchRandomQuote,
  },
)(FetchRandomQuoteButton);
