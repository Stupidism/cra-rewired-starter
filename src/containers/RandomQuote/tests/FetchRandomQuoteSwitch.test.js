import React from 'react';
import { shallow } from 'enzyme';

import { FetchRandomQuoteButton } from '../FetchRandomQuoteButton';

describe('<FetchRandomQuoteButton />', () => {
  it('should be consistent with snapshot', () => {
    const snapshot = shallow(
      <FetchRandomQuoteButton loading>Button</FetchRandomQuoteButton>,
    );
    expect(snapshot).toMatchSnapshot();
  });
});
