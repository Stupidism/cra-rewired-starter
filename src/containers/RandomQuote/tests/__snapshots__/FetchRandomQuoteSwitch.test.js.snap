// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FetchRandomQuoteButton /> should be consistent with snapshot 1`] = `
<Button
  ghost={false}
  loading={true}
  prefixCls="ant-btn"
  type="primary"
>
  Button
</Button>
`;
